boot.js

```diff
@@ -27,14 +27,22 @@ function wrap (server, opts, instance) {
     throw new Error(readyKey + '() is already defined, specify an expose option')
   }
 
-  server[useKey] = function (a, b, c) {
-    instance.use(a, b, c)
-    return this
+  server[useKey] = function (fn, opts) {
+    instance.use(fn, opts)
+    const plugin = instance._lastUsed
+    const thenableDescriptor = {
+      then: {
+        value (resolve) {
+          return instance.assimilate(plugin).then(resolve)
+        }
+      }
+    }
+    return Object.create(this, thenableDescriptor)
```

Comment:
I think this is not documented right now.

Reply:
https://github.com/mcollina/avvio/pull/92/files#diff-04c6e90faac2675aa89e2176d2eec7d8R148

Reply:
https://github.com/mcollina/avvio/pull/92/files#diff-04c6e90faac2675aa89e2176d2eec7d8R167

Reply:
"thenanble wrapped instance" should suffice?